data-runtime/src/main/java/io/micronaut/data/runtime/operations/internal/sql/DefaultSqlPreparedQuery.java

```diff
@@ -327,6 +356,9 @@ public List<Cursor> createCursors(List<Object> results, Pageable pageable, Runti
         if (pageable.getMode() == Mode.CURSOR_PREVIOUS) {
             Collections.reverse(results);
         }
+        RuntimePersistentEntity<E> persistentEntity = getPersistentEntity();
```

Comment:
`runtimePersistentEntity` is already passed and used below, not sure if this is needed and it looks confusing

Reply:
Fixed